Home/Home 9.1/MsgPackSerializer.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+
+class MsgPackSerializer {
+    
+    static func serialize(_ data: MyDataType) -> [UInt8] {
+        var buffer: [UInt8] = []
+        
+        switch data.getValue() {
+        case .int16(let value):
+            serializeInt16(&buffer, value)
+        case .string(let value):
+            serializeString(&buffer, value)
+        case .map(let value):
+            serializeMap(&buffer, value)
+        }
+        
+        return buffer
+    }
+    
+    static func deserialize(_ data: [UInt8]) -> MyDataType {
+        var index = 0
+        return deserializeValue(data, &index)
+    }
+    
+    private static func serializeInt16(_ buffer: inout [UInt8], _ value: Int16) {
+        buffer.append(0xd1)  // int16
+        buffer.append(UInt8((value >> 8) & 0xFF))
+        buffer.append(UInt8(value & 0xFF))
+    }
+    
+    private static func serializeString(_ buffer: inout [UInt8], _ value: String) {
+        guard value.count <= 30 else {
```

Comment:
should it be 31? and you have check below related to 32. I hope you should cleanup this code